bson/src/test/unit/org/bson/vector/VectorGenericBsonTest.java

```diff
@@ -0,0 +1,272 @@
+ * limitations under the License.
+ */
+
+package org.bson.vector;
+
+import org.bson.BsonArray;
+import org.bson.BsonBinary;
+import org.bson.BsonDocument;
+import org.bson.BsonString;
+import org.bson.BsonValue;
+import org.bson.Float32Vector;
+import org.bson.PackedBitVector;
+import org.bson.Vector;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+import util.JsonPoweredTestHelper;
+
+import java.io.File;
+import java.io.IOException;
+import java.net.URISyntaxException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.stream.Stream;
+
+import static java.lang.String.format;
+import static org.bson.BsonHelper.decodeToDocument;
+import static org.bson.BsonHelper.encodeToHex;
+import static org.bson.internal.vector.VectorHelper.determineVectorDType;
+import static org.junit.Assert.assertThrows;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.junit.jupiter.api.Assumptions.assumeFalse;
+
+// BSON tests powered by language-agnostic JSON-based tests included in test resources
+class VectorGenericBsonTest {
+
+    private static final List<String> TEST_NAMES_TO_IGNORE = Arrays.asList(
+            //NO API to set padding for Floats available
+            "FLOAT32 with padding",
+            //NO API to set padding for Floats available
+            "INT8 with padding",
+            //It is impossible to provide float inputs for INT8 in the API
+            "INT8 with float inputs",
+            //It is impossible to provide float inputs for INT8
+            "Underflow Vector PACKED_BIT",
+            //It is impossible to provide float inputs for PACKED_BIT in the API
+            "Vector with float values PACKED_BIT",
+            //It is impossible to provide float inputs for INT8
+            "Overflow Vector PACKED_BIT",
+            "Overflow Vector INT8",
+            // It is impossible to provide -129 for byte.
+            "Underflow Vector INT8");
+
+
+    @ParameterizedTest(name = "{0}")
+    @MethodSource("provideTestCases")
+    void shouldPassAllOutcomes(@SuppressWarnings("unused") final String description,
+                               final BsonDocument testDefinition, final BsonDocument testCase) {
+        assumeFalse(TEST_NAMES_TO_IGNORE.contains(testCase.get("description").asString().getValue()));
+
+        String testKey = testDefinition.getString("test_key").getValue();
+        boolean isValidVector = testCase.getBoolean("valid").getValue();
+        if (isValidVector) {
+            runValidTestCase(testKey, testCase);
+        } else {
+            runInvalidTestCase(testCase);
+        }
+    }
+
+    private void runInvalidTestCase(final BsonDocument testCase) {
```

Comment:
Let's make `static` this and all the other methods in `VectorGenericBsonTest` except for `shouldPassAllOutcomes`.